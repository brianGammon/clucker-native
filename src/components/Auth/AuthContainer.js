/* @flow */
import * as React from 'react';
import { ImageBackground } from 'react-native';
import {
  View, Text, Container, Content, H2,
} from 'native-base';
import AuthHeader from './AuthHeader';
import styles from './styles';
import type AuthExtraLink from './AuthExtraLink';

type Props = {
  title: string,
  error: string,
  children: any,
  extraLinks: AuthExtraLink[],
};

const AuthBase = ({
  title, error, children, extraLinks,
}: Props) => (
  <Container>
    <ImageBackground
      source={require('../../assets/rays.jpg')}
      style={styles.imageBackground}
    >
      <Content contentContainerStyle={styles.container}>
        <View padder style={styles.body}>
          <AuthHeader />
          <H2 style={styles.title}>{title}</H2>
          {error && <Text style={styles.error}>{error}</Text>}
          <View style={styles.formContainer}>{children}</View>
          {extraLinks
            && extraLinks.length > 0 && (
              <View padder>{extraLinks.map(link => link)}</View>
          )}
        </View>
      </Content>
    </ImageBackground>
  </Container>
);

export default AuthBase;
