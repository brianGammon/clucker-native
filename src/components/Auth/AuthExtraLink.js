/* @flow */
import React from 'react';
import { View, Text, Button } from 'native-base';
import styles from './styles';

type Props = {
  message: string,
  buttonText: string,
  handlePress: () => void,
};

const AuthExtraLink = ({ message, buttonText, handlePress }: Props) => (
  <View style={styles.extraLinks}>
    <Text style={styles.bodyText}>{message}</Text>
    <Button transparent style={{ alignSelf: 'center' }} onPress={handlePress}>
      <Text>{buttonText}</Text>
    </Button>
  </View>
);

export default AuthExtraLink;
