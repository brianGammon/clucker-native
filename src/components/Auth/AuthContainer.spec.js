import React from 'react';
import AuthContainer from './AuthContainer';
import AuthExtraLink from './AuthExtraLink';

describe('AuthContainer', () => {
  const link1 = (
    <AuthExtraLink
      key={0}
      message="test"
      buttonText="text"
      handlePress={() => {}}
    />
  );
  const link2 = (
    <AuthExtraLink
      key={1}
      message="test"
      buttonText="text"
      handlePress={() => {}}
    />
  );
  test('renders without extra links and no error', () => {
    const wrapper = shallow(<AuthContainer title="Sign In" />);
    expect(wrapper).toMatchSnapshot();
  });

  test('renders with extra links and no error', () => {
    const wrapper = shallow(
      <AuthContainer title="Sign In" extraLinks={[link1, link2]} />,
    );
    expect(wrapper).toMatchSnapshot();
  });

  test('renders with empty extra links and error', () => {
    const wrapper = shallow(
      <AuthContainer
        title="Sign In"
        extraLinks={[]}
        error="Test error message"
      />,
    );
    expect(wrapper).toMatchSnapshot();
  });
});
