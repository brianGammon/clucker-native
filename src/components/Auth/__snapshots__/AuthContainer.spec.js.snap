// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthContainer renders with empty extra links and error 1`] = `
<Styled(Container)>
  <ImageBackground
    source={
      Object {
        "testUri": "../../../src/assets/rays.jpg",
      }
    }
    style={
      Object {
        "height": "100%",
        "width": "100%",
      }
    }
  >
    <Styled(Content)
      contentContainerStyle={
        Object {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
          "minHeight": 500,
        }
      }
    >
      <Styled(ViewNB)
        padder={true}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "rgba(255, 255, 255, .85)",
            "borderRadius": 20,
            "maxWidth": 450,
            "width": "90%",
          }
        }
      >
        <AuthHeader />
        <Styled(H2)
          style={
            Object {
              "margin": 5,
            }
          }
        >
          Sign In
        </Styled(H2)>
        <Styled(Text)
          style={
            Object {
              "color": "red",
              "padding": 20,
            }
          }
        >
          Test error message
        </Styled(Text)>
        <Styled(ViewNB)
          style={
            Object {
              "width": "100%",
            }
          }
        />
      </Styled(ViewNB)>
    </Styled(Content)>
  </ImageBackground>
</Styled(Container)>
`;

exports[`AuthContainer renders with extra links and no error 1`] = `
<Styled(Container)>
  <ImageBackground
    source={
      Object {
        "testUri": "../../../src/assets/rays.jpg",
      }
    }
    style={
      Object {
        "height": "100%",
        "width": "100%",
      }
    }
  >
    <Styled(Content)
      contentContainerStyle={
        Object {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
          "minHeight": 500,
        }
      }
    >
      <Styled(ViewNB)
        padder={true}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "rgba(255, 255, 255, .85)",
            "borderRadius": 20,
            "maxWidth": 450,
            "width": "90%",
          }
        }
      >
        <AuthHeader />
        <Styled(H2)
          style={
            Object {
              "margin": 5,
            }
          }
        >
          Sign In
        </Styled(H2)>
        <Styled(ViewNB)
          style={
            Object {
              "width": "100%",
            }
          }
        />
        <Styled(ViewNB)
          padder={true}
        >
          <AuthExtraLink
            buttonText="text"
            handlePress={[Function]}
            key="0"
            message="test"
          />
          <AuthExtraLink
            buttonText="text"
            handlePress={[Function]}
            key="1"
            message="test"
          />
        </Styled(ViewNB)>
      </Styled(ViewNB)>
    </Styled(Content)>
  </ImageBackground>
</Styled(Container)>
`;

exports[`AuthContainer renders without extra links and no error 1`] = `
<Styled(Container)>
  <ImageBackground
    source={
      Object {
        "testUri": "../../../src/assets/rays.jpg",
      }
    }
    style={
      Object {
        "height": "100%",
        "width": "100%",
      }
    }
  >
    <Styled(Content)
      contentContainerStyle={
        Object {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
          "minHeight": 500,
        }
      }
    >
      <Styled(ViewNB)
        padder={true}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "rgba(255, 255, 255, .85)",
            "borderRadius": 20,
            "maxWidth": 450,
            "width": "90%",
          }
        }
      >
        <AuthHeader />
        <Styled(H2)
          style={
            Object {
              "margin": 5,
            }
          }
        >
          Sign In
        </Styled(H2)>
        <Styled(ViewNB)
          style={
            Object {
              "width": "100%",
            }
          }
        />
      </Styled(ViewNB)>
    </Styled(Content)>
  </ImageBackground>
</Styled(Container)>
`;
